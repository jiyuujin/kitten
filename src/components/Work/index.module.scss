.subtitle {
    font-size: 1.2rem;
    margin: 18px 0;
}

.description {
    font-size: 0.8rem;
    line-height: 24px;

    a {
        color: #dd66a1;
    }

    ul {
        margin: 0;
    }
}
