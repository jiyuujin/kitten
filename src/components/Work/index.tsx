import React, { FC } from 'react'
import { useIntl } from 'gatsby-plugin-intl'

type WorkProps = {
  data: Array<unknown>
}

const Work: FC<WorkProps> = ({ data }) => {
  const intl = useIntl()

  return (
    <div className="section">
      <h1>{intl.formatMessage({ id: 'work' })}</h1>
      <p>{intl.formatMessage({ id: 'work_experience' })}</p>
      <h2>{intl.formatMessage({ id: 'labels.related_links' })}</h2>
      <ul>
        {data.map(({ node }: any) => {
          return (
            <li key={node.id}>
              <a
                href={node.url}
                role="button"
                aria-pressed="true"
                target="_blank"
                rel="noopener noreferrer"
              >
                {intl.formatMessage({
                  id: node.text,
                })}
              </a>
            </li>
          )
        })}
      </ul>
    </div>
  )
}

export default Work
