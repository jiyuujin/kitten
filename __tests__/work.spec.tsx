import React from 'react'
import renderer from 'react-test-renderer'
import Work from '../src/components/Work'

jest.mock('react-intl', () => {
    const reactIntl = jest.requireActual('react-intl')
    const intl = reactIntl.createIntl({
        locale: 'en',
    })
    return {
        ...reactIntl,
        useIntl: () => intl,
    }
})

describe('Work', () => {
    it('render correctly', () => {
        const tree = renderer.create(<Work />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
