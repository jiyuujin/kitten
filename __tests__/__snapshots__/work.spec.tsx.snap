// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work render correctly 1`] = `
<div>
  work
  <div>
    work_experience
  </div>
  <div>
    related_links
  </div>
  <div>
    <ul>
      <li>
        <a
          href="https://github.com/jiyuujin/Curriculum-Vitae"
          rel="noopener noreferrer"
          target="_blank"
        >
          curriculum_vitae
        </a>
      </li>
      <li>
        <a
          href="https://scrapbox.io/nekohack/skillset"
          rel="noopener noreferrer"
          target="_blank"
        >
          skill_set_coming_soon
        </a>
      </li>
    </ul>
  </div>
  <div>
    skill
  </div>
  <div>
    <div>
      <h2>
        Vue.js & Nuxt.js
      </h2>
      <div>
        v-kansai team member ‚ô•Ô∏èÔ∏è
      </div>
      <div
        className=""
      >
        Vue.js Advocator üòé
      </div>
    </div>
    <div>
      <h2>
        Node.js & JavaScript
      </h2>
      <div>
        ES6
      </div>
      <div>
        Plugin Creation
      </div>
      <div
        className=""
      >
        Babel
      </div>
      <div
        className=""
      >
        Webpack
      </div>
    </div>
    <div>
      <h2>
        React
      </h2>
      <div>
        React Enthusiast üòÄ
      </div>
      <div
        className=""
      >
        Next.js
      </div>
      <div
        className=""
      >
        Gatsby.js
      </div>
    </div>
    <div>
      <h2>
        VC & CI
      </h2>
      <div>
        Git
      </div>
      <div
        className=""
      >
        Github Actions
      </div>
      <div
        className=""
      >
        Gitlab CI
      </div>
      <div
        className=""
      >
        Travis CI
      </div>
    </div>
    <div>
      <h2>
        Infrastructure
      </h2>
      <div>
        Serverless ‚úçÔ∏è
      </div>
      <div>
        AWS
      </div>
      <div
        className=""
      >
        GCP / Firebase
      </div>
      <div
        className=""
      >
        Netlify
      </div>
    </div>
    <div>
      <h2>
        Scala
      </h2>
      <div>
        Playframework
      </div>
      <div
        className=""
      >
        Akka HTTP
      </div>
    </div>
  </div>
</div>
`;
